nixos/modules/services/backup/duplicity.nix

```diff
@@ -0,0 +1,99 @@
+{ config, lib, pkgs, ...}:
+
+with lib;
+
+let cfg = config.services.duplicity;
+
+in {
+  options.services.duplicity = {
+    enable = mkEnableOption "backups with duplicity";
+
+    root = mkOption {
+      type = types.path;
+      default = "/";
+      description = ''
+        Root directory to backup.
+      '';
+    };
+
+    include = mkOption {
+      type = types.listOf types.path;
+      default = [];
+      example = [ "/home" ];
+      description = ''
+        List of paths to include into the backups. See
+        <citerefentry><refentrytitle>duplicity</refentrytitle>
+        <manvolnum>1</manvolnum></citerefentry> for details on the syntax.
+      '';
+    };
+
+    exclude = mkOption {
+      type = types.listOf types.path;
+      default = [];
+      description = ''
+        List of paths to exclude from backups. See
+        <citerefentry><refentrytitle>duplicity</refentrytitle>
+        <manvolnum>1</manvolnum></citerefentry> for details on the syntax.
+      '';
+    };
+
+    targetUrl = mkOption {
+      type = types.str;
+      example = "s3://host:port/prefix";
+      description = ''
+        Target url to backup to. See
+        <citerefentry><refentrytitle>duplicity</refentrytitle>
+        <manvolnum>1</manvolnum></citerefentry> for supported urls.
+      '';
+    };
+
+    secretFile = mkOption {
+      type = types.nullOr types.path;
+      default = null;
+      description = ''
+        Path of a file containing secrets (gpg passphrase, access key...). It
+        is sourced before duplicity is run. For example, a file containing:
+          export PASSPHRASE='...'
+          export AWS_ACCESS_KEY_ID='...'
+          export AWS_SECRET_ACCESS_KEY='...'
```

Comment:
Wrap this with the `<programlisting>` xml attribute to make it look correctly in the docs (build them yourself to check).